src/Integration/Binding/BindingController.cs

```diff
@@ -92,59 +91,44 @@ protected override int OnQueryStatus(ref Guid pguidCmdGroup, uint cCmds, OLECMD[
             return base.OnQueryStatus(ref pguidCmdGroup, cCmds, prgCmds, pCmdText);
         }
 
-        private bool OnBindStatus(ProjectViewModel projectVM)
+        private bool OnBindStatus(BindCommandArgs args)
         {
-            return this.OnBindStatus(projectVM?.Project);
-        }
-
-        private void OnBind(ProjectViewModel projectVM)
-        {
-            this.OnBind(projectVM?.Project);
-        }
-
-        private bool OnBindStatus(SonarQubeProject projectInformation)
-        {
-            return projectInformation != null
+            return args != null
+                && args.ProjectKey != null
                 && this.host.VisualStateManager.IsConnected
                 && !this.host.VisualStateManager.IsBusy
                 && VsShellUtils.IsSolutionExistsAndFullyLoaded()
                 && VsShellUtils.IsSolutionExistsAndNotBuildingAndNotDebugging()
                 && (this.projectSystemHelper.GetSolutionProjects()?.Any() ?? false);
         }
 
-        private void OnBind(SonarQubeProject projectInformation)
+        private void OnBind(BindCommandArgs args)
         {
-            Debug.Assert(this.OnBindStatus(projectInformation));
+            Debug.Assert(this.OnBindStatus(args));
 
             var componentModel = host.GetService<SComponentModel, IComponentModel>();
             TelemetryLoggerAccessor.GetLogger(componentModel)?.ReportEvent(TelemetryEvent.BindCommandCommandCalled);
 
-            this.workflowExecutor.BindProject(projectInformation);
+            SonarQubeProject projectInformation = new SonarQubeProject(args.ProjectKey, args.ProjectName);
+
+            this.workflowExecutor.BindProject(projectInformation, args.Connection);
         }
 
         #endregion
 
         #region IBindingWorkflowExecutor
 
-        void IBindingWorkflowExecutor.BindProject(SonarQubeProject projectInformation)
+        void IBindingWorkflowExecutor.BindProject(SonarQubeProject projectInformation, ConnectionInformation connection)
         {
             //TODO: CM2 - choose the type of binding
-            var workflow = CreateBindingWorkflow(projectInformation);
+            var workflow = new BindingWorkflow(this.host, connection, projectInformation);
 
             IProgressEvents progressEvents = workflow.Run();
             Debug.Assert(progressEvents != null, "BindingWorkflow.Run returned null");
-            this.SetBindingInProgress(progressEvents, projectInformation);
-        }
-
-        private IBindingWorkflow CreateBindingWorkflow(SonarQubeProject projectInformation)
-        {
-            ConnectionInformation connection = this.host.VisualStateManager.GetConnectedServer(projectInformation);
```

Comment:
I've changed the binding so that the caller needs to pass in all of the required information i.e. the connection information as well as project information.
Previously, the code triggering the binding only passed in the project instance, and this class called back out to the host to get the connection information. It's now the responsibility of the caller to pass in all of the required information.